src/merkle/path.rs

```diff
@@ -190,67 +277,67 @@ impl MerklePathSet {
 
         Ok(())
     }
-}
-
-// HELPER FUNCTIONS
-// --------------------------------------------------------------------------------------------
 
-fn is_even(pos: u64) -> bool {
-    pos & 1 == 0
-}
+    // HELPER FUNCTIONS
+    // --------------------------------------------------------------------------------------------
 
-/// Calculates the hash of the parent node by two sibling ones
-/// - node — current node
-/// - node_pos — position of the current node
-/// - sibling — neighboring vertex in the tree
-fn calculate_parent_hash(node: Word, node_pos: u64, sibling: Word) -> Word {
-    if is_even(node_pos) {
-        Rpo256::merge(&[node.into(), sibling.into()]).into()
-    } else {
-        Rpo256::merge(&[sibling.into(), node.into()]).into()
+    const fn is_even(pos: u64) -> bool {
+        pos & 1 == 0
     }
-}
-
-/// Returns vector of hashes from current to the root
-fn compute_path_trace(path: &[Word], depth: u32, index: u64) -> (Vec<Word>, Word) {
-    let mut pos = 2u64.pow(depth) + index;
 
-    let mut computed_hashes = Vec::<Word>::new();
+    /// Returns hash of the root
+    fn compute_path_root(path: &[Word], depth: u32, index: u64) -> Word {
+        let mut pos = 2u64.pow(depth) + index;
 
-    let mut comp_hash = Rpo256::merge(&[path[0].into(), path[1].into()]).into();
+        // hash that is obtained after calculating the current hash and path hash
+        let mut comp_hash = Rpo256::merge(&[path[0].into(), path[1].into()]).into();
 
-    if path.len() != 2 {
         for path_hash in path.iter().skip(2) {
-            computed_hashes.push(comp_hash);
             pos /= 2;
-            comp_hash = calculate_parent_hash(comp_hash, pos, *path_hash);
+            comp_hash = Self::calculate_parent_hash(comp_hash, pos, *path_hash);
         }
+
+        comp_hash
     }
```

Comment:
How different is this from `compute_root()` I proposed above? Could we replace this function with `MerklePath::compute_root()`?

Reply:
I was avoiding refactoring other structs to keep this PR succinct. There is a `tech-debt introduced` covering this case.

The main difference is that for `MerklePathSet`, the input node is the first element of the merkle path, and `MerklePath` will take that as `verify` argument. I think the latter approach is cleaner as we should separate one from the other, but we probably need to think of the consequences for the advice provider.

Reply:
Makes sense!